src/main/java/com/externalSorting/app/ExternalSortingRunner.java

```diff
@@ -9,60 +9,74 @@
     private final String originalFile;
     private final String outputFolder;
 
-    public ExternalSortingRunner(String originalFile, String outputFile) {
+    /**
+     * Constructor
+     *
+     * @param originalFile
+     * Path to the file that will be externally sorted
+     * @param outputFolder
+     * Path to the folder which will contain both the temporary files and the output file
+     */
+    public ExternalSortingRunner(String originalFile, String outputFolder) {
         this.originalFile = originalFile;
-        this.outputFolder = outputFile;
+        this.outputFolder = outputFolder;
     }
 
-    protected void externalSort() {
-
-        System.out.println(outputFolder + "/tmpdir");
-        File newDirectory = new File("src/main/resources/tmpdir/");
-        List<File> files = readAndSplitData("src/main/resources/SmallLibrary.txt", newDirectory);
-
-        System.out.println(outputFolder + "/Output.txt");
-        mergeFiles(files, new File("src/main/resources/SmallLibraryOutput.txt"), comparator);
+    /**
+     * This method is meant to be called from main function.
+     * It is responsible for:
+     *  creating a temporary folder which will hold data files
+     *  calling readAndSplitData and mergeFiles
+     *  deleting temporary folder and files after the process is finished
+     */
+    public void externalSort() {
+        File newDirectory = new File(outputFolder + "/tmpdir");
+        newDirectory.mkdir();
+        List<File> files = readAndSplitData(originalFile, newDirectory);
+        mergeFiles(files, new File(outputFolder + "/Output.txt"));
         for (File file : files) {
             file.delete();
         }
         newDirectory.delete();
     }
+
     /**
-    CHANGE_FILE
-    Maximum number of lines of text that can be put into a single temporary file
+    TMP_FILE_MAXSIZE
+    Maximum number of bytes of a single temporary file. The value of this variable controls memory usage.
     */
-    private static final int TMP_FILE_MAXSIZE = 24000000; // 24mb
+    private static final int TMP_FILE_MAXSIZE = 10000000; // around 10MB
```

Comment:
We haven't discussed it but maybe it's worth to make this value dynamic i.e. set it's value depending on maximum memory. Reason why I suggest this is that if someone runs the program with -Xmx8mb, it will crash so in order to ensure the stability, consider dynamic calculation of TMP_FILE_MAXSIZE depending on available memory.